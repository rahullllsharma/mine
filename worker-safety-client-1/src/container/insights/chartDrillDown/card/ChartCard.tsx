import type { ReactNode } from "react";
import { Icon } from "@urbint/silica";
import { useTenantStore } from "@/store/tenant/useTenantStore.store";

type ChartCardProps = {
  title: string;
  type: "hazard" | "control";
  chart?: ReactNode;
};

const EmptyCardContent = ({ type }: { type: string }): JSX.Element => {
  const { hazard, control } = useTenantStore(state => state.getAllEntities());

  const typeLabel =
    type === "hazard"
      ? hazard.label.toLowerCase()
      : control.label.toLowerCase();
  return (
    <>
      <p className="text-neutral-shade-75 text-sm px-4 text-center">
        {`A chart will appear here when one (1) ${typeLabel} is selected`}
      </p>
      <Icon
        name="bar_chart_square"
        className="text-neutral-shade-58 text-4xl"
      />
    </>
  );
};

export default function ChartCard({
  title,
  type,
  chart = <EmptyCardContent type={type} />,
}: ChartCardProps): JSX.Element {
  return (
    <div className="flex flex-col items-center">
      <p className="text-neutral-shade-100 text-base font-semibold mb-2">
        {title}
      </p>
      <div className="flex flex-col items-center justify-center border border-dashed border-neutral-shade-58 rounded-lg w-full h-96">
        {chart}
      </div>
    </div>
  );
}
