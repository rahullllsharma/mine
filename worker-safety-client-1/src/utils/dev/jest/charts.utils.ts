import type { RiskLevelByDate } from "@/components/charts/dailyRiskHeatmap/types";
import { act } from "@testing-library/react";
import { RiskLevel } from "@/components/riskBadge/RiskLevel";

export function chartRenderHelper(): () => void {
  // set up a listener so we can force the width/height
  // this is required to get the chart to render in its ResponsiveWrapper
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  let listener: ((rect: any) => void) | undefined;
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  (window as any).ResizeObserver = class ResizeObserver {
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    constructor(ls: any) {
      listener = ls;
    }
    // eslint-disable-next-line @typescript-eslint/no-empty-function
    observe() {}
    // eslint-disable-next-line @typescript-eslint/no-empty-function
    disconnect() {}
  };

  function triggerResize(): void {
    act(() => {
      listener &&
        listener([
          {
            contentRect: {
              x: 0,
              y: 0,
              width: 800,
              height: 800,
              top: 100,
              bottom: 0,
              left: 100,
              right: 0,
            },
          },
        ]);
    });
  }
  return triggerResize;
}

export const sampleHeatmapDates = {
  startDate: "2022-01-29",
  endDate: "2022-02-03",
};

// consumed by the DailyRiskHeatmap
export const sampleRiskLevelByDate: RiskLevelByDate[] = [
  {
    date: "2022-01-29",
    riskLevel: RiskLevel.HIGH,
  },
  {
    date: "2022-01-30",
    riskLevel: RiskLevel.MEDIUM,
  },
  {
    date: "2022-01-31",
    riskLevel: RiskLevel.UNKNOWN,
  },
  {
    date: "2022-02-01",
    riskLevel: RiskLevel.LOW,
  },
  {
    date: "2022-02-02",
    riskLevel: RiskLevel.RECALCULATING,
  },
  {
    date: "2022-02-03",
    riskLevel: RiskLevel.UNKNOWN,
  },
];
