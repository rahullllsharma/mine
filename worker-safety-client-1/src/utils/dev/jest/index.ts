export * from "./intl.utils";
export * from "./testing.utils";
export * from "./charts.utils";
