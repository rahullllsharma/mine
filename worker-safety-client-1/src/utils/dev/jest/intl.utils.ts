/** Mock the default locale when using the Intl.DateTimeFormat class */
export function mockIntlDateTimeFormatLocale(locale: string): void {
  const { DateTimeFormat } = Intl;
  jest
    .spyOn(global.Intl, "DateTimeFormat")
    .mockImplementation((_, options) => new DateTimeFormat(locale, options));
}
