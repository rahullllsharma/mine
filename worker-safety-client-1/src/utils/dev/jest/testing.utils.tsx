/* istanbul ignore file */
import type { ReactNode } from "react";
import type { PropsWithClassName } from "@/types/Generic";
import { fireEvent, render, screen } from "@testing-library/react";
import { FormProvider, useForm } from "react-hook-form";
import { useTenantStore } from "@/store/tenant/useTenantStore.store";
import { TenantMock } from "@/store/tenant/utils/tenantMock";

export const formTemplate = (
  component: ReactNode,
  defaultValues: { [key: string]: any } = {}
) => {
  function Wrapper({ children }: PropsWithClassName): JSX.Element {
    const methods = useForm({ defaultValues });
    return <FormProvider {...methods}>{children}</FormProvider>;
  }
  return <Wrapper>{component}</Wrapper>;
};

export const formRender = (
  component: ReactNode,
  defaultValues: { [key: string]: any } = {}
) => {
  return render(formTemplate(component, defaultValues));
};

// to overcome react-select limitation for testing, of not responding to click events
export const openSelectMenu = (role = "combobox", name?: RegExp): void => {
  const inputElement = screen.getByRole(role, { name });
  fireEvent.keyDown(inputElement, { key: "ArrowDown", code: "ArrowDown" });
};

export const mockTenantStore = (mock = TenantMock) => {
  const { setTenant } = useTenantStore.getState();
  setTenant(mock);
};
