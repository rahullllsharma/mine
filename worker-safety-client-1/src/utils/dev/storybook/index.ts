export * from "./WrapperForm.utils";
export * from "./WrapperModal.utils";
export * from "./WrapperScroll.utils";
