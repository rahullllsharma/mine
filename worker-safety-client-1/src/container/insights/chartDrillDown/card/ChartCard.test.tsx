import { render, screen } from "@testing-library/react";
import { mockTenantStore } from "@/utils/dev/jest";
import ChartCard from "./ChartCard";

describe(ChartCard.name, () => {
  mockTenantStore();
  describe("when it renders", () => {
    it("should match the snapshot", () => {
      const { asFragment } = render(
        <ChartCard title="lorem ipsum" type="hazard" />
      );
      expect(asFragment()).toMatchSnapshot();
    });

    it('should have a text element based on the "title" prop', () => {
      render(<ChartCard title="lorem ipsum" type="hazard" />);
      screen.getByText(/lorem ipsum/i);
    });
  });

  describe('when a react node is passed in the "charts" prop', () => {
    it("should render the node", () => {
      render(
        <ChartCard
          title="lorem ipsum"
          type="hazard"
          chart={<div>This is a chart</div>}
        />
      );
      screen.getByText(/this is a chart/i);
    });
  });
});
