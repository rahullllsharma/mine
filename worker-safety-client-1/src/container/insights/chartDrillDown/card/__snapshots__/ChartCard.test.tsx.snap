// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChartCard when it renders should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col items-center"
  >
    <p
      class="text-neutral-shade-100 text-base font-semibold mb-2"
    >
      lorem ipsum
    </p>
    <div
      class="flex flex-col items-center justify-center border border-dashed border-neutral-shade-58 rounded-lg w-full h-96"
    >
      <p
        class="text-neutral-shade-75 text-sm px-4 text-center"
      >
        A chart will appear here when one (1) hazard is selected
      </p>
      <i
        aria-hidden="true"
        class="ci-bar_chart_square text-neutral-shade-58 text-4xl"
      />
    </div>
  </div>
</DocumentFragment>
`;
